### ISSUE TITLE
💅 false positive of `useLiteralKeys` with `process.env["TEST_VARIABLE"]`

### ISSUE THREAD

**Issue by BenjaminBrienen** [biomejs/biome#11015]
### Environment information

<details>

```bash

CLI:
  Version:                      2.5.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:                npm
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.17.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/11.13.1

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0

```

</details>

### Rule name

useLiteralKeys

### Playground link

https://biomejs.dev/playground/?lintRules=useLiteralKeys&tab=formatter&pane=Diagnostics&code=ZABlAGwAZQB0AGUAIABwAHIAbwBjAGUAcwBzAC4AZQBuAHYAWwAiAFQARQBTAFQAXwBWAEEAUgBJAEEAQgBMAEUAIgBdAA%3D%3D

### Expected result

This lint should not trigger because the "fix" is an error.

```
Property 'TEST_VARIABLE' comes from an index signature, so it must be accessed with ['TEST_VARIABLE'].
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
